### DISCUSSION TITLE
ImportError: DLL load failed while importing _ctypes: The specified module could not be found.

### DISCUSSION THREAD

**Discussion by misunderstoodreamer** [pyinstaller/pyinstaller#7431] [Help]
Hello to all,

I have a desktop application created with PyQT5 library. In order to give the application, I need to take the exe. When pandas library imported, exe throws me error called "ImportError: DLL load failed while importing _ctypes: The specified module could not be found." If I don't use pandas lib, exe will be created successfully. I have already tried to reinstall Anaconda3 and I use venv. Also I have tried different interpretors. How will I handle the problem? Any help will be appreciated.

![image](https://user-images.githubusercontent.com/74912434/217566321-fb56e9b3-65dd-4cfe-a3f4-45199b11d096.png)

**Comment by bwoodsend** [comment]
Don't use anaconda with PyInstaller. This is exactly the kind of nonsense you should expect to happen when you do. Install a normal build of Python from https://python.org/downloads and install PyInstaller and your code in that.

**Comment by rokm** [comment]
Looks like the `_ctypes` extension failed to load due to a missing DLL; which is either `libffi-7.dll`, `python3x.dll`, or `vcruntim140.dll`. Check if these are collected into your frozen application.

**Comment by bersbersbers** [comment]
Hi,
I have exactly the same problem. Here's my stack trace (of what I call `new.exe`):

```
Traceback (most recent call last):
  File "C:\Users\ContainerAdministrator\scoop\persist\python311\Lib\site-packages\PyInstaller\hooks\rthooks\pyi_rth_mplconfig.py", line 45, in <module>
  File "C:\Users\ContainerAdministrator\scoop\persist\python311\Lib\site-packages\PyInstaller\hooks\rthooks\pyi_rth_mplconfig.py", line 31, in _pyi_rthook
  File "PyInstaller\loader\pyimod02_importers.py", line 391, in exec_module
  File "PyInstaller\fake-modules\_pyi_rth_utils\__init__.py", line 23, in <module>
  File "PyInstaller\loader\pyimod02_importers.py", line 391, in exec_module
  File "PyInstaller\fake-modules\_pyi_rth_utils\_win32.py", line 12, in <module>
  File "PyInstaller\loader\pyimod02_importers.py", line 391, in exec_module
  File "ctypes\__init__.py", line 8, in <module>
ImportError: DLL load failed while importing _ctypes: The specified module could not be found.
```

As you see, I build this executable on a private GitLab CI, using a `mcr.microsoft.com/windows/server:ltsc2022` image into which I install `scoop` followed by `python311`. I am no (aware of) using Anaconda in any way.

Interestingly, I recently switched to this image and way of installing Python from using `python:3.11` images, from which the executable (`old.exe`) would easily run, so I looked into the differences.

First, when I copy `libffi-8.dll` next to `new.exe`, it does run. This is not reuired in `old.exe`. Now,

> Looks like the _ctypes extension failed to load due to a missing DLL; which is either libffi-7.dll, python3x.dll, or vcruntim140.dll. Check if these are collected into your frozen application.

I did. In fact, it seems to be `libffi-8.dll` now, but the interesting point is that `old.exe` has `libffi-8.dll`, `libcrypto-3.dll`, `and libssl-3.dll` in the MEI folder, while `new.exe` have the same files (plus `tcl86.t.dll` and `tk86t.dll`) in a `dlls` subfolder of the MEI folder. So `libffi-8.dll` *is* bundled with the application, but in a different location than in `old.exe`.

Is that a `PyInstaller` bug? Is that something I can easily fix? Or is `scoop` doing some magic that confuses `PyInstaller`? I should add that I test my application on the same CI before building it, and there, everything seems to run fine.

Here's a comparison of the MEI folders using `windiff`:
![image](https://github.com/pyinstaller/pyinstaller/assets/12128514/a09a9b3f-1237-4795-a3e7-b7fdd99b99e0)


(`pyprojec.toml` and `metadata` differ since I relaxed a requirement on `PySide6-Essentials`, but since `pyi_rth_mplconfig.py` is related to `matplotlib`, I don't think this plays a role here. Also, I cannot reproduce the problem using the same build commands on the same `scoop`/`python311` locally.)

**Reply by bersbersbers** [reply]
I should add that I can reproduce the problem locally now. The difference was in my use of a `.venv` as well:
- Install `scoop`, then `scoop install python311`, then `pip install pyinstaller`, etc.: fails
- Install `scoop`, then `scoop install python311`, then create and activate a `.venv`, then `pip install pyinstaller`, etc.: works!

**Reply by bersbersbers** [reply]
I filed https://github.com/pyinstaller/pyinstaller/issues/8023
